test/test_time_stepping.jl

```diff
@@ -39,52 +39,52 @@ end
     This test ensures that when we compute w from the continuity equation that the full velocity field
     is divergence-free.
 """
-function test_compute_w_from_continuity(arch, ft)
+function compute_w_from_continuity(arch, FT)
     Nx, Ny, Nz = 16, 16, 16
     Lx, Ly, Lz = 16, 16, 16
 
-    grid = RegularCartesianGrid(ft, (Nx, Ny, Nz), (Lx, Ly, Lz))
+    grid = RegularCartesianGrid(FT, (Nx, Ny, Nz), (Lx, Ly, Lz))
 
-    u = FaceFieldX(arch, grid)
-    v = FaceFieldY(arch, grid)
-    w = FaceFieldZ(arch, grid)
-    div_u = CellField(arch, grid)
+    u = FaceFieldX(FT, arch, grid)
+    v = FaceFieldY(FT, arch, grid)
+    w = FaceFieldZ(FT, arch, grid)
+    div_u = CellField(FT, arch, grid)
 
-    u.data .= rand(Nx, Ny, Nz)
-    v.data .= rand(Nx, Ny, Nz)
+    @views data(u) .= rand(FT, Nx, Ny, Nz)

```

Comment:
I'm not 100% sure that this actually sets `u`, but perhaps you verified it? One way to avoid this problem is to have `data` return a view directly (which is also nice because it doesn't allocate [much] memory), as noted earlier in the review

Reply:
You're right. It wasn't actually setting `u` but test was still passing as the `w` computed from continuity was also zero, and thus the velocity was divergence-free lol.

`data` now returns a view so this has been fixed.